daemon/api.go

```diff
@@ -778,6 +781,7 @@ func snapInstall(inst *snapInstruction, st *state.State) (string, []*state.TaskS
 }
 
 func snapUpdate(inst *snapInstruction, st *state.State) (string, []*state.TaskSet, error) {
+	// TODO: bail if revision is given (and != current?), *or* behave as with install --revision?
```

Comment:
That's the case we discussed in the sprint. refresh of a particular revision should be allowed if the revision is locally available. It should behave just like any other refresh.

If it's not locally available, then yes, it might behave as install in the sense that it works as long as the snap is personal.

The basic semantics of install and refresh should be preserved though: install does not work if the snap is already installed. Refresh doesn't work if the snap is not installed.